lib/in_out/reader.py

```diff
@@ -22,60 +24,111 @@ def linux_input(log_directory, channels_requested, starting_date, ending_date):
         logs (dict) : dictionary with key(str) as 'yyyy-mm-dd' and value as another dictionary {"data":day_data,"channel":channel_name}
 
     """
-    def check_if_path_and_file_is_valid(work_path, channels_requested):
-        if not os.path.exists(work_path):
-            raise IOError("Path " + work_path + " doesn't exist")
-
-        all_channels = set()
-        for channel in os.listdir(work_path):
-            all_channels.add(channel[:-4])
-        #if user has passed the channels_requested argument as "ALL", we make the channels_requested equal to all the available channels in work_path directory
-        if len(channels_requested) == 1 and channels_requested[0] == "ALL":
-            return list(all_channels)
-
-        channel_list = []
-
-        for channel in channels_requested:
-            if channel not in all_channels:
-                print "[Error | io/linuxInput] Channel " + work_path + channel + ".txt doesn't exist"
-            else:
-                channel_list.append(channel)
-        return channel_list
-
-
+    logs = defaultdict(list)
+    reader = UbuntuReader()
```

Comment:
The class that is instantiated depends on the channel(s) requested. Hence it might be better to have a factory method design pattern that takes the channel name and gives an object of matching reader class. I suggest the following changes to the code.

* Reader - abstract class for all readers
* UbuntuReader, SlackReader, ScummVMReader - concrete instances of Reader abstract class
* ReaderFactory - abstract class to create readers based on channel name
* LinuxReaderFactory - concrete class to create concrete channel-specific readers that work on Linux OS.  In this PR, this just creates the right Reader object and returns the same. We can move the Linux specific parts of Reader class to this factory in the next PR.The <channel name, reader class> mapping can be stored in a dictionary that is read from an INI file.


### References
1. Factory method pattern, pages 106 - 117, Design Patterns: Elements of Reusable Object-Oriented Software, Eric Gamma et al., 1997.
1. [Wikipedia](https://en.wikipedia.org/wiki/Factory_method_pattern), 
1. [Python example](https://krzysztofzuraw.com/blog/2016/factory-pattern-python.html)
1. [Read INI file using configparser](https://stackoverflow.com/questions/3220670/read-all-the-contents-in-ini-file-into-dictionary-with-python)